worker/internal/benthos/transformers/transform_character_substitution.go

```diff
@@ -48,52 +49,57 @@ Substituted: Ifmmp Xpsme 234@%^
 Note that this does not work for hex values: 0x00 -> 0x1F
 */
 
-func TransformCharacterSubstitution(value string) (*string, error) {
+func TransformCharacterScramble(value string) (*string, error) {
 
-	transformedString := strings.Map(substituteChar, value)
+	transformedString := strings.Map(ScrambleChar, value)
 
 	return &transformedString, nil
 }
 
-func substituteChar(r rune) rune {
-	letterMap := map[rune]rune{
-		'a': 'b', 'b': 'c', 'c': 'd', 'd': 'e', 'e': 'f', 'f': 'g', 'g': 'h', 'h': 'i', 'i': 'j',
-		'j': 'k', 'k': 'l', 'l': 'm', 'm': 'n', 'n': 'o', 'o': 'p', 'p': 'q', 'q': 'r', 'r': 's',
-		's': 't', 't': 'u', 'u': 'v', 'v': 'w', 'w': 'x', 'x': 'y', 'y': 'z', 'z': 'a',
-	}
+func ScrambleChar(r rune) rune {
+	letterList := "abcdefghijklmnopqrstuvwxyz"
 
-	numberMap := map[rune]rune{
-		'0': '1', '1': '2', '2': '3', '3': '4', '4': '5', '5': '6', '6': '7', '7': '8', '8': '9', '9': '0',
-	}
+	numberList := "0123456789"
+
+	specialCharList := "!@#$%^&*()-+=_ []{}|\\ ;\"<>,./?"
 
-	specialCharMap := map[rune]rune{
-		'!': '@', '@': '#', '#': '$', '$': '%', '%': '^', '^': '&', '&': '*',
-		'*': '(', '(': ')', ')': '-', '-': '+', '+': '=', '=': '_',
-		'_': '[', '[': ']', ']': '{', '{': '}', '}': '|', '|': '\\',
-		'\\': ':', ':': ';', ';': '\'', '\'': '"', '"': '<', '<': '>',
-		'>': ',', ',': '.', '.': '/', '/': '?', '?': '!',
+	randStringListInd, err := transformer_utils.GenerateRandomInt64InValueRange(0, int64(len(letterList))-1)
+	if err != nil {
+		return r
 	}
 
 	if unicode.IsLetter(r) {
-		lowerR := unicode.ToLower(r)
-		sub, ok := letterMap[lowerR]
-		if ok {
-			if unicode.IsUpper(r) {
-				return unicode.ToUpper(sub)
-			}
-			return sub
+
+		sub := rune(letterList[randStringListInd])
+		if unicode.IsUpper(r) {
+			return unicode.ToUpper(sub)
 		}
+		return sub
+
+	}
+
+	randNumberListInd, err := transformer_utils.GenerateRandomInt64InValueRange(0, int64(len(numberList))-1)
+	if err != nil {
+		return r
 	}
 
 	if unicode.IsDigit(r) {
-		sub, ok := numberMap[r]
-		if ok {
-			return sub
-		}
+
+		return rune(numberList[randNumberListInd])
 	}
 
-	if sub, ok := specialCharMap[r]; ok {
-		return sub
+	randInd, err := transformer_utils.GenerateRandomInt64InValueRange(0, int64(len(specialCharList))-1)
+	if err != nil {
+		return r
+	}
+
+	if transformer_utils.IsAllowedSpecialChar(r) {
+		if unicode.IsSpace(r) {
```

Comment:
little Q here, but should this check happen before anything else? Seems like we alway want to return the space in a string, where as right now we only check if it's a space if it's an allowed special char.

could the logic flow of this function be:

```
if unicode.IsLetter(r) {}
else if unicode.IsSpace(r) {}
else if unicode.IsNumber(r) {}
else if IsAllowedSpecialChar(r) {}
```

Reply:
refactored to htis